internal/impl/kafka/franz_reader.go

```diff
@@ -73,6 +75,45 @@ const (
 	TransactionIsolationLevelReadCommitted TransactionIsolationLevel = "read_committed"
 )
 
+// consumeResetOffsetType describes the offset to start consuming from, or if OffsetOutOfRange is seen while fetching,
+// to restart consuming from.
+type consumeResetOffsetType string
+
+const (
+	// consumeResetOffsetAtStart corresponds to auto.offset.reset "earliest"
+	consumeResetOffsetAtStart consumeResetOffsetType = "at_start"
+	// consumeResetOffsetAtEnd corresponds to auto.offset.reset "latest"
+	consumeResetOffsetAtEnd consumeResetOffsetType = "at_end"
+	// consumeResetOffsetAtCommitted corresponds to auto.offset.reset "none"
+	consumeResetOffsetAtCommitted consumeResetOffsetType = "at_committed"
+)
+
+const (
+	// FranzConsumerFieldLintRules contains the lint rules for the consumer fields.
+	FranzConsumerFieldLintRules = `
+let has_topic_partitions = this.topics.any(t -> t.contains(":"))
+
+root = [
+  if $has_topic_partitions {
+    if this.consumer_group.or("") != "" {
+      "this input does not support both a consumer group and explicit topic partitions"
+    } else if this.regexp_topics {
+      "this input does not support both regular expression topics and explicit topic partitions"
+    }
+  } else {
+    if this.consumer_group.or("") == "" {
+      "a consumer group is mandatory when not using explicit topic partitions"
+    }
+  },
+  # We don't have any way to distinguish between start_from_oldest set explicitly to true and not set at all, so we
+  # assume users will be OK if consume_reset_offset overwrites it silently
```

Comment:
Can we change it to be an `Optional` property and use `this.exists` instead?

Reply:
Unfortunately, I'd have to remove the default value of `start_from_oldest` for that to work, which is `true` rather than `false` and that would break existing configs... I guess I could set it to `true` if the field is not present in the config, but that would be confusing. Not sure if that makes sense. WDYT?